lib/extensions/ar_slug.rb

```diff
@@ -0,0 +1,13 @@
+module ArSlug
+  extend ActiveSupport::Concern
+
+  included do
+    virtual_column :href_slug, :type => :string
```

Comment:
after discussing with @Fryguy, we're keeping it as href_slug as it's the most relevant/natural name w.r.t. slug for the href attribute, as compared to href and miq_href_slug.

In a separate PR though (replication), we should check/enforce that no href or href_slug attributes are named as such as it would clash with the API.